manifests/install.pp

```diff
@@ -4,37 +4,41 @@
 class candlepin::install {
   assert_private()
 
-  $enable_pki_core = $facts['os']['release']['major'] == '8'
+  if !$candlepin::use_container {
+    $enable_pki_core = $facts['os']['release']['major'] == '8'
 
-  if $candlepin::java_package {
-    stdlib::ensure_packages([$candlepin::java_package])
-    Package[$candlepin::java_package] -> Package['candlepin']
-  }
-
-  if $enable_pki_core {
-    package { 'pki-core':
-      ensure      => installed,
-      enable_only => true,
-      provider    => 'dnfmodule',
-      before      => Package['candlepin'],
+    if $candlepin::java_package {
+      stdlib::ensure_packages([$candlepin::java_package])
+      Package[$candlepin::java_package] -> Package['candlepin']
     }
-  }
 
-  package { ['candlepin']:
-    ensure => $candlepin::version,
-  }
+    if $enable_pki_core {
+      package { 'pki-core':
+        ensure      => installed,
+        enable_only => true,
+        provider    => 'dnfmodule',
+        before      => Package['candlepin'],
+      }
+    }
 
-  if $facts['os']['selinux']['enabled'] {
-    package { ['candlepin-selinux']:
+    package { ['candlepin']:
       ensure => $candlepin::version,
     }
 
-    if $enable_pki_core {
-      Package['pki-core'] -> Package['candlepin-selinux']
+    if $facts['os']['selinux']['enabled'] {
+      package { ['candlepin-selinux']:
+        ensure => $candlepin::version,
+      }
+
+      if $enable_pki_core {
+        Package['pki-core'] -> Package['candlepin-selinux']
+      }
     }
-  }
 
-  if $candlepin::run_init {
-    stdlib::ensure_packages(['wget'], { ensure => $candlepin::wget_version, })
+    if $candlepin::run_init {
+      stdlib::ensure_packages(['wget'], { ensure => $candlepin::wget_version, })
+    }
```

Comment:
You dropped `cpinit` from `service.pp` so can be removed if we drop cpinit functionality.